endpoint/endpoint.go

```diff
@@ -276,6 +292,47 @@ func (e *Endpoint) GetLoggerConfig() log.LoggerConfig {
 	return e.config.LoggerConfig
 }
 
+// CreateOrUpdateMetrics uses DataRegistryService CreateOrUpdateMetrics to create or update metrics
+func (e *Endpoint) CreateOrUpdateMetrics(ctx context.Context, metricWrappers []*data_receiver.MetricWrapper) ([]*RegisterResponse, error) {
+	if e.config.APIKey != "" {
+		ctx = metadata.AppendToOutgoingContext(ctx, APIKeyHeader, e.config.APIKey)
+	}
+	stream, err := e.regclient.CreateOrUpdateMetrics(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, metricWrapper := range metricWrappers {
+		request := &zproto.RegisterMetricRequest{
+			Metric:     metricWrapper,
+			UpdateMode: zproto.UpdateMode_REPLACE, // replace by default
+		}
+		if err := stream.Send(request); err != nil {
+			log.Log(e, log.LevelError, log.Fields{
+				"metric wrapper": metricWrapper,
+			}, "Error sending metric wrapper to data-registry")
+			return nil, err
+		}
+	}
+	registerResponse, err := stream.CloseAndRecv()
+	if err != nil {
+		log.Log(e, log.LevelError, log.Fields{}, "Error closing stream")
+		return nil, err
+	}
+
+	result := make([]*RegisterResponse, 0, len(metricWrappers))
+	for _, response := range registerResponse.Responses {
+		if response.Error != "" {
+			result = append(result, &RegisterResponse{Error: errors.New(response.Error)})
+		} else {
+			result = append(result, &RegisterResponse{
+				MetricID:   response.Response.InstanceId,
+				MetricName: response.Response.Name,
+			})
+		}
+	}
+	return result, nil
```

Comment:
The other data_receiver calls are tracking some stats, what would make sense to track here?

Reply:
Trying failed/succeeded registration count